docs/deploy-rbd.md

```diff
@@ -205,6 +210,34 @@ secrets under `encryptionPassphrase` key and switch `encrypted` option in
 StorageClass to `"true"`. This is not supported for storage classes that already
 have PVs provisioned.
 
+### Encryption KMS configuration
+
+To further improve security robustness it is possible to use unique passphrases
+generated for each volume and stored in a Key Management System (KMS). Currently
+HashiCorp Vault is the only KMS supported.
+
+To use Vault as KMS set `encryptionKMS` to `vault` and `encryptionKMSID` to a
+unique identifier for Vault configuration. You will also need to create vault
+configuration similar to the [example](../examples/rbd/kms-config.yaml)
+and use same `encryptionKMSID`. In order for ceph-csi to be able to access the
+configuration you will need to have it mounted to csi-rbdplugin containers as
+`ceph-csi-encryption-kms-config` config map.
+
+#### Configuring HashiCorp Vault
+
+Using Vault as KMS you need to configure Kubernetes authentication method as
+described in [official
+documentation](https://www.vaultproject.io/docs/auth/kubernetes.html).
+
+If token reviewer is used, you will need to configure service account for
+that also like in [example](../examples/rbd/csi-vaulttokenreview-rbac.yaml) to
+be able to review jwt tokens.
+
+Configure a role(s) for service accounts used for ceph-csi:
+
+* provisioner requires only delete permissions
+* attacher requires create and read permissions
```

Comment:
create should happen at CreateVolume right? if you want to support encryption per PVC?

Reply:
we do not encrypt volumes on creation but rather on first time attaching it (same way as we create filesystem there), so that's why we need "create" permission in vault for service account used in attachment process

Reply:
below are the 2 scenarios am trying to understand
 as the creation is happening in the NodeStageVolume request

1) if you are creating RWX PVC so you will be creating multiple keys if it is scheduled in multiple nodes
2) if the pod is moved from one node to another node again you will be creating a new key to encrypt it?

Reply:
it should not create a new key as we store it by volumeID
and because process is:
- try to get key
- if key not found:
  - generate new key
  - save it in kms
if it fails before it saved and get rescheduled - the new key will be generated, but once it's saved it should be reused